config/webpack/shared.js

```diff
@@ -56,16 +56,18 @@ module.exports = {
 
     new ManifestPlugin({
       publicPath: output.publicPath,
-      writeToFileEmit: true
-    })
+      writeToFileEmit: true,
+    }),
+    new webpack.optimize.CommonsChunkPlugin({
+      name: 'vendor',
+    }),
   ],
 
   resolve: {
     extensions: settings.extensions,
-    modules: [
-      resolve(settings.source_path),
-      'node_modules'
-    ]
+    modules: [resolve(settings.source_path)].concat(
+      Object.values(engines).map(engine => `${engine}/node_modules`)
+    ),
```

Comment:
one concern that was raised was if the same package is included from both core and a plugin/engine, which one wins.
I've added a syntax error to my plugin/node_modules/react/index.js file and confirmed that it does include the react pacakge from core vs the one from the plugin/node_modules directory.

Reply:
With some more experimenting, looks like this behaves fine for directly referenced dependencies, but if you don't explicitly require it, you end up with the wrong one.

(Requiring something that requires react in a plugin pack which depends on 0.15 will end up with 0.16 , but if you add that explicit `import * as React from 'react'`, you get the right version.)

Not necessarily a bug, but something to watch out for with dependencies.